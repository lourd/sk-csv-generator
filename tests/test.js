const path = require('path')
const fs = require('mz/fs')

const { run } = require('../')

const testFiles = ['animals', 'dogs', 'pokemon']

describe('test files', () => {
  testFiles.forEach(input => {
    const inFile = `${input}.csv`
    const outFile = `${input}-out.csv`
    it(`${inFile} should transform to ${outFile}`, async () => {
      const file = path.resolve(__dirname, inFile)
      const expectedFile = path.resolve(__dirname, outFile)
      const [outBuf, results] = await Promise.all([
        fs.readFile(expectedFile),
        run(file)
      ])
      expect(results).toBe(outBuf.toString())
    })
  })

  it('should throw an error if a property does not have a "has official name" property', async () => {
    const file = path.resolve(__dirname, './no-name.csv')
    expect.assertions(1)
    try {
      await run(file)
    } catch (e) {
      expect(e).toMatchSnapshot()
    }
  })
})
