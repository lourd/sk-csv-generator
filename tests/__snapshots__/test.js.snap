// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test files should throw an error if a property does not have a "has official name" property 1`] = `
[Error: Missing "has official name" column for item: {
  "inherits": "animal",
  "is a": "dog",
  "prefix": "dog",
  "official name": "Labrador",
  "type": "working"
}]
`;
